daemon/mgr/container.go

```diff
@@ -1629,43 +1629,49 @@ func (mgr *ContainerManager) markStoppedAndRelease(c *Container, m *ctrd.Message
 // exitedAndRelease be register into ctrd as a callback function, when the running container suddenly
 // exited, "ctrd" will call it to set the container's state and release resouce and so on.
 func (mgr *ContainerManager) exitedAndRelease(id string, m *ctrd.Message) error {
-	// update container info
 	c, err := mgr.container(id)
 	if err != nil {
 		return err
 	}
 
-	c.Lock()
-	defer c.Unlock()
-
 	if err := mgr.markStoppedAndRelease(c, m); err != nil {
 		return err
 	}
 
-	c.State.Status = types.StatusExited
+	c.SetStatusExited()
+
+	// Action Container Remove and function exitedAndRelease are conflict.
+	// If a container has been removed and the corresponding meta.json will be removed as well.
+	// However, when this function exitedAndRelease still keeps the container instance,
+	// there will be possibility that in exitedAndRelease, code calls c.Write(mgr.Store) to
+	// write the removed meta.json again. If that, incompatibilty happens.
+	// As a result, we check whether this container is still in the meta store.
+	if container, err := mgr.container(c.Name); err != nil || container == nil {
```

Comment:
This block of code fixes pull request https://github.com/alibaba/pouch/issues/1332. @fuweid 
Since in the ci test, the issue would not happen.